modules/core/shared/src/main/scala/io/circe/CursorOp.scala

```diff
@@ -1,47 +1,49 @@
 package io.circe
 
 import cats.{ Eq, Show }
+import cats.syntax.all._
 import java.io.Serializable
 
 sealed abstract class CursorOp extends Product with Serializable {
+  import CursorOp._
 
   /**
    * Does this operation require the current focus (not context) to be an
    * object?
    */
-  def requiresObject: Boolean
+  final def requiresObject: Boolean =
+    this match {
+      case _: DownField =>
+        true
+      case _ =>
+        false
+    }
```

Comment:
I have to wonder if this is better than the original?  Is the method dispatch slower than the type check in the pattern match?  I know its easier to understand whats going on, but is the trade off worth it?

Reply:
TL;DR I'm totally fine moving this to the object.

That said, I'd be even more fine just removing it, except I'd like to minimize bincompat changes. It is only used by the old FailedCursor, which the work in this PR replaces. https://github.com/circe/circe/blob//modules/core/shared/src/main/scala/io/circe/FailedCursor.scala#L11

It is used to help describe the failure condition, which is now handled much more clearly by the CursorFailureReason error.

Reply:
Since the use case in circe would be deprecated by this method and is not in performance critical code, do you still want me to move it?

Reply:
FYI, my reason for moving it was just to try to make this file less verbose.